src/datasets/vision/imagenet.jl

```diff
@@ -0,0 +1,171 @@
+  "class_names"       => Vector{SubString{String}}[["tench", "Tinca tinca"], ["goldfish", "…
+  "wnid_to_label"     => Dict("n07693725"=>932, "n03775546"=>660, "n01689811"=>45, "n021008…
+
+julia> dataset.metadata["class_names"][y]
+  3-element Vector{SubString{String}}:
+   "common iguana"
+   "iguana"
+   "Iguana iguana"
+```
+
+# References
+
+[1]: [Russakovsky et al., ImageNet Large Scale Visual Recognition Challenge](https://arxiv.org/abs/1409.0575)
+"""
+struct ImageNet <: SupervisedDataset
+    metadata::Dict{String,Any}
+    split::Symbol
+    dataset::FileDataset
+    targets::Vector{Int}
+    inverse_preprocess::Function
+end
+
+ImageNet(; split=:train, Tx=Float32, kws...) = ImageNet(Tx, split; kws...)
+ImageNet(split::Symbol; kws...) = ImageNet(; split, kws...)
+ImageNet(Tx::Type; kws...) = ImageNet(; Tx, kws...)
+
+function ImageNet(
+    Tx::Type,
+    split::Symbol;
+    preprocess=ImageNetReader.default_preprocess,
+    inverse_preprocess=ImageNetReader.default_inverse_preprocess,
+    dir=nothing,
+    train_dir="train",
+    val_dir="val",
+    test_dir="test",
+    devkit_dir="devkit",
+)
+    @assert split ∈ (:train, :val, :test)
+
+    DEPNAME = "ImageNet"
+    METADATA_FILENAME = joinpath(devkit_dir, "data", "meta.mat")
+
+    TRAINSET_SIZE = 1_281_167
+    VALSET_SIZE = 50_000
+    TESTSET_SIZE = 100_000
+
+    # Load metadata
+    file_path = datafile(DEPNAME, METADATA_FILENAME, dir)
+    metadata = ImageNetReader.read_metadata(file_path)
+
+    root_dir = @datadep_str DEPNAME
+    if split == :train
+        dataset = ImageNetReader.get_file_dataset(
+            Tx, preprocess, joinpath(root_dir, train_dir)
+        )
+        @assert length(dataset) == TRAINSET_SIZE
+    elseif split == :val
+        dataset = ImageNetReader.get_file_dataset(
+            Tx, preprocess, joinpath(root_dir, val_dir)
+        )
+        @assert length(dataset) == VALSET_SIZE
+    else
+        dataset = ImageNetReader.get_file_dataset(
+            Tx, preprocess, joinpath(root_dir, test_dir)
+        )
+        @assert length(dataset) == TESTSET_SIZE
+    end
+    targets = [
+        metadata["wnid_to_label"][wnid] for wnid in ImageNetReader.get_wnids(dataset)
+    ]
+    return ImageNet(metadata, split, dataset, targets, inverse_preprocess)
+end
+
+convert2image(d::ImageNet, x::AbstractArray) = d.inverse_preprocess(x)
+
+Base.length(d::ImageNet) = length(d.dataset)
+
+const IMAGENET_MEM_WARNING = """Loading the entire ImageNet dataset into memory might not be possible.
+    If you are sure you want to load all of ImageNet, use `dataset[1:end]` instead of `dataset[:]`.
+    """
+Base.getindex(::ImageNet, ::Colon) = throw(ArgumentError(IMAGENET_MEM_WARNING))
+Base.getindex(d::ImageNet, i::Integer) = (features=d.dataset[i], targets=d.targets[i])
+function Base.getindex(d::ImageNet, is::AbstractVector)
+    return (features=StackView(d.dataset[is]), targets=d.targets[is])
+end
```

Comment:
```suggestion
function Base.getindex(d::ImageNet, is::AbstractVector)
    return (features=StackView(d.dataset[is]), targets=d.targets[is])
end
```
Let's remove this method entirely and avoid the `StackView` dependency. Collation can be performed by downstream consumers, e.g. dataloaders

Reply:
Removed in fe38d43.